### PR TITLE
fix readme

### PR DESCRIPTION
**Pull request by littletomatodonkey**
polish cv reprod doc

### CODE REVIEW

File: docs/ThesisReproduction_CV.md [hunk 1]
```diff
@@ -179,7 +133,7 @@ AlexNet-Prod项目提供了基于reprod_log的5个验收点对齐验收示例，
 
 **【实战】**
 
-对于AlexNet网络结构的PyTorch实现 [alexnet-pytorch](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/AlexNet-torch/torchvision/models/alexnet.py)，[alexnet-paddle](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/AlexNet-paddle/paddlevision/models/alexnet.py) 提供了网络结构代码转换为PaddlePaddle的实现。
+对于AlexNet网络结构的PyTorch实现 [alexnet-pytorch](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/pipeline/Step1/AlexNet_torch/torchvision/models/alexnet.py)，[alexnet-paddle](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/pipeline/Step1/AlexNet_paddle/paddlevision/models/alexnet.py) 提供了网络结构代码转换为PaddlePaddle的实现。
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 这里改为下面这样好一点

AlexNet网络结构的PyTorch实现: [alexnet-pytorch](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/pipeline/Step1/AlexNet_torch/torchvision/models/alexnet.py)
对应转换后的PaddlePaddle实现: [alexnet-paddle](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/pipeline/Step1/AlexNet_paddle/paddlevision/models/alexnet.py)

File: docs/ThesisReproduction_CV.md [hunk 2]
```diff
@@ -224,31 +178,45 @@ def transfer():
...
 
 #### 3.1.3 模型组网正确性验证
 
 **【基本流程】**
 
-    * 定义PyTorch模型，加载权重，固定seed，基于numpy生成随机数，转换为PyTorch可以处理的tensor，送入网络，获取输出，使用reprod_log保存结果。
-    * 定义PaddlePaddle模型，加载权重，固定seed，基于numpy生成随机数，转换为PaddlePaddle可以处理的tensor，送入网络，获取输出，使用reprod_log保存结果。
-    *  使用reprod_log排查diff，小于阈值，即可完成自测。
-    
+* 定义PyTorch模型，加载权重，固定seed，基于numpy生成随机数，转换为PyTorch可以处理的tensor，送入网络，获取输出，使用reprod_log保存结果。
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 使用有序列表步骤清楚一点

File: docs/ThesisReproduction_CV.md [hunk 3]
```diff
@@ -224,31 +178,45 @@ def transfer():
...
 [https://github.com/littletomatodonkey/AlexNet-Prod/tree/master/pipeline/Step1](https://github.com/littletomatodonkey/AlexNet-Prod/tree/master/pipeline/Step1)
 
-### 3.2 验证/测试集数据读取对齐
 
-#### 3.2.1 数据集类Dataset复现方法
+**【验收】**
+
+对于待复现的项目，前向对齐验收流程如下。
+
+* 准备输入：fake data
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 这里也用有序列表

File: docs/ThesisReproduction_CV.md [hunk 4]
```diff
@@ -224,31 +178,45 @@ def transfer():
...
+    * 使用参考代码的dataloader，生成一个batch的数据，保存下来，在前向对齐时，直接从文件中读入。
+    * 固定随机数种子，生成numpy随机矩阵，转化tensor
+* 保存输出：
+    * PaddlePaddle/PyTorch：dict，key为tensor的name（自定义），value为tensor的值。最后将dict保存到文件中。建议命名为`forward_paddle.npy`和`forward_pytorch.npy`。
+* 自测：使用reprod_log加载2个文件，使用report功能，记录结果到日志文件中，建议命名为`forward_diff_log.txt`，观察diff，二者diff小于特定的阈值即可。
+* 提交内容：新建文件夹，将`forward_paddle.npy`、`forward_pytorch.npy`与`forward_diff_log.txt`文件放在文件夹中，后续的输出结果和自查日志也放在该文件夹中，一并打包上传即可。
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 打包上传，这里需要说明上传地址吗

File: docs/ThesisReproduction_CV.md (hunk 1 above)
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 这里改为下面这样会好一点

AlexNet网络结构的PyTorch实现:  [alexnet-pytorch](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/pipeline/Step1/AlexNet_torch/torchvision/models/alexnet.py)
转换为PaddlePaddle的实现: [alexnet-paddle](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/pipeline/Step1/AlexNet_paddle/paddlevision/models/alexnet.py)

File: docs/ThesisReproduction_CV.md (hunk 2 above)
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 这里用有序列表

File: docs/ThesisReproduction_CV.md (hunk 3 above)
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 这里用有序列表

File: docs/ThesisReproduction_CV.md [hunk 5]
```diff
@@ -261,324 +229,193 @@ AlexNet模型组网正确性验证可以参考如下示例代码：
...
 
 ### 3.3 评估指标对齐
 
-#### 3.3.1 复现方法
+**【基本流程】**
 
 PaddlePaddle提供了一系列Metric计算类，比如说`Accuracy`, `Auc`, `Precision`, `Recall`等，而PyTorch中，目前可以通过组合的方式实现metric计算，或者调用[torchmetrics](https://torchmetrics.readthedocs.io/en/latest/)，在论文复现的过程中，需要注意保证对于该模块，给定相同的输入，二者输出完全一致。
 
 * 操作流程
     * 定义PyTorch模型，加载训练好的权重（需要是官网repo提供好的），获取评估结果，使用reprod_log保存结果。
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 流程使用有序列表

File: docs/ThesisReproduction_CV.md [hunk 6]
```diff
@@ -261,324 +229,193 @@ AlexNet模型组网正确性验证可以参考如下示例代码：
...
 
-输出如下
+对于待复现的项目，评估指标对齐验收流程如下。
 
-```
-2021-09-27 11:19:48,955 - reprod_log.utils - INFO - top1:
-2021-09-27 11:19:48,955 - reprod_log.utils - INFO -     mean diff: check passed: True, value: 0.0
-2021-09-27 11:19:48,955 - reprod_log.utils - INFO - diff check passed
-```
+* 输入：dataloader, model
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 流程使用有序列表

File: docs/ThesisReproduction_CV.md [hunk 7]
```diff
@@ -261,324 +229,193 @@ AlexNet模型组网正确性验证可以参考如下示例代码：
...
 
 PaddlePaddle与PyTorch均提供了很多loss function，用于模型训练，具体的API映射表可以参考：[Loss类API映射列表](https://www.paddlepaddle.org.cn/documentation/docs/zh/guides/08_api_mapping/pytorch_api_mapping_cn.html#lossapi)。以CrossEntropyLoss为例，主要区别为：
 * PaddlePaddle提供了对软标签、指定softmax计算纬度的支持。
 
 如果论文中使用的loss function没有指定的API，则可以尝试通过组合API的方式，实现自定义的loss function。
 
 * 操作流程
-    * 定义PyTorch模型，加载权重，固定seed，基于numpy生成随机数，转换为PyTorch可以处理的tensor，送入网络，获取loss结果，使用reprod_log保存结果。
-    * 定义PaddlePaddle模型，加载权重，固定seed，基于numpy生成随机数，转换为PaddlePaddle可以处理的tensor，送入网络，获取loss结果，使用reprod_log保存结果。
+    * 定义PyTorch模型，加载权重，加载fake data 和 fake label（或者固定seed，基于numpy生成随机数），转换为PyTorch可以处理的tensor，送入网络，获取loss结果，使用reprod_log保存结果。
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 流程使用有序列表

File: docs/ThesisReproduction_CV.md [hunk 8]
```diff
@@ -261,324 +229,193 @@ AlexNet模型组网正确性验证可以参考如下示例代码：
...
-2021-09-27 11:29:40,692 - reprod_log.utils - INFO -     mean diff: check passed: True, value: 9.5367431640625e-07
-2021-09-27 11:29:40,692 - reprod_log.utils - INFO - diff check passed
-```
+本部分可以参考文档：[https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/pipeline/Step3/README.md](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/pipeline/Step3/README.md)。
+
+**【验收】**
+
+对于待复现的项目，损失函数对齐验收流程如下。
+
+* 输入：fake data & label
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 流程使用有序列表

File: docs/ThesisReproduction_CV.md [hunk 9]
```diff
@@ -261,324 +229,193 @@ AlexNet模型组网正确性验证可以参考如下示例代码：
...
 ### 3.8 反向对齐
 
-#### 3.8.1 基本步骤
+**【基本流程】**
 
-模型前向对齐之后，复现完优化器、损失函数等模块之后，需要进行模型反向对齐。
+此处可以通过numpy生成假的数据和label（推荐），也可以准备固定的真实数据。具体流程如下。
 
-此处可以通过numpy生成假的数据和label（推荐），也可以准备固定的真实数据。
+* 检查两个代码的训练超参数全部一致，如优化器及其超参数、学习率、BatchNorm/LayerNorm中的eps等。
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 流程使用有序列表

File: docs/ThesisReproduction_CV.md [hunk 10]
```diff
@@ -261,324 +229,193 @@ AlexNet模型组网正确性验证可以参考如下示例代码：
...
-```
+**【实战】**
+
+本部分可以参考文档：[反向对齐操作文档](https://github.com/littletomatodonkey/AlexNet-Prod/blob/master/pipeline/Step4/README.md#%E5%8F%8D%E5%90%91%E5%AF%B9%E9%BD%90%E6%93%8D%E4%BD%9C%E6%96%B9%E6%B3%95)。
+
+**【验收】**
+
+对于待复现的项目，反向对齐验收流程如下。
+
+* 输入：fake data & label
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md]: 流程使用有序列表

File: docs/ThesisReproduction_CV.md [hunk 11]
```diff
@@ -609,20 +452,50 @@ random.seed(config.SEED)
...
+**【基本流程】**
 
-* 操作流程
-    * 准备train/eval data, loader, model
-    * 对model按照论文所述进行初始化(如果论文中提到加载pretrain，则按需加载pretrained model)
-    * 加载配置，开始训练，迭代得到最终模型与评估指标，将评估指标使用reprod_log保存到文件中。
-    * 将PaddlePaddle提供的参考指标使用reprod_log提交到另一个文件中。
-    * 使用reprod_log排查diff，小于阈值，即可完成自测。
+完成前面的步骤之后，就可以开始全量数据的训练对齐任务了。按照下面的步骤进行训练对齐。
+
```
Comment (WenmuZhou) [inline docs/ThesisReproduction_CV.md:539]: 流程使用有序列表

File: docs/ThesisReproduction_CV.md (hunk 1 above)
Comment (littletomatodonkey) [reply docs/ThesisReproduction_CV.md]: done

File: docs/ThesisReproduction_CV.md (hunk 2 above)
Comment (littletomatodonkey) [reply docs/ThesisReproduction_CV.md]: done, thx

File: docs/ThesisReproduction_CV.md (hunk 3 above)
Comment (littletomatodonkey) [reply docs/ThesisReproduction_CV.md]: done, thx

File: docs/ThesisReproduction_CV.md (hunk 4 above)
Comment (littletomatodonkey) [reply docs/ThesisReproduction_CV.md]: 这里不需要，一般参加比赛，会提示他们发邮件提交结果的

File: docs/ThesisReproduction_CV.md (hunk 11 above)
Comment (littletomatodonkey) [reply docs/ThesisReproduction_CV.md:539]: done, thx

**Review by dyning — APPROVED** [review]
